mk/disable-api-docs.mk

```diff
@@ -0,0 +1,7 @@
+# This file is active unless `./configure --enable-api-docs`.
+# Running `make api-html` would indicate a mistake in the caller.
+
+.PHONY: api-html
+api-html:
+	@echo "API Docs are disabled. Configure with '--enable-api-docs', or avoid calling 'make api-html'."
```

Comment:
Just for convenience to give the user a better error, modeled after what @roberth did for the tests when they are disabled.

Reply:
We could inline these files into the main Makefile that imports them, but it'd clutter up the main Makefile a bit. I think it's nice to keep that one clean.
I expect these files to be essentially write once and read very very rarely.